src/GameLogic/Entities/Fireball.ts

```diff
@@ -0,0 +1,83 @@
+import {Entity} from '../Entity';
+import {EntityType, FloorType} from '../Enums';
+import {Direction8, Direction8Utils} from '../../../src.common/Enums/Direction8';
+import {Level} from '../Level';
+import {Protagonist} from './Protagonist';
+
+export class Fireball implements Entity {
+	public readonly type: EntityType;
+
+	public x: number;
+
+	public y: number;
+
+	public direction: Direction8;
+
+	constructor(direction: Direction8) {
+		this.type = EntityType.Fireball;
+		this.x = 0;
+		this.y = 0;
+		this.direction = direction;
+	}
+
+	public update(level: Level): void {
+		// Try moving forward
+		if (this.isMoveAllowed(level, this.direction)) {
```

Comment:
What do you think about removing the returns and turning it into if-elseif?

```
if (...) {
    move();
elseif (...) { 
    move();
} elseif ...
```

Reply:
Yeah, this is probably better